### PR TITLE
Adding CategoricalImputer documentation

### PR DESCRIPTION
**Pull request by aatallah**
Please review this new entry for the categorical imputer. Comments/suggestions/improvements welcomed! :smile:

### CODE REVIEW

File: feature-engineering/categorical_imputer.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
+Impute missing categorical values using reference features.
```
Comment (znation) [inline feature-engineering/categorical_imputer.md]: Grammatically this is a command (telling someone -- the user? -- to do something). Should we say "The Feature Engineering toolkit can impute..." or "Using this toolkit, you can impute..."?

Comment (aatallah) [inline feature-engineering/categorical_imputer.md]: I've followed the example from numeric_imputer.md. Happy to change it.

Comment (znation) [inline feature-engineering/categorical_imputer.md]: I'm reviewing this in isolation (I don't think I reviewed that one, or if I did, I was less thorough). Please feel free to keep things consistent (consistency is probably better than having better grammar in one place).

File: feature-engineering/categorical_imputer.md [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+The imputer takes as an input a column which contains categorical values, and may contain
+some None values. Its secondary input is a set of reference feature columns. During the 
+Fit phase, the imputer learns the relationships between the values of the column to be
```
Comment (znation) [inline feature-engineering/categorical_imputer.md]: Are these phases part of the user interaction surface of the imputer (does the user have fine-grained control over phases)? If not, can we abstract this concept away from the user and just say "The imputer learns the relationships between the values of the column to be imputed and the reference feature columns, then applies that learned relationship to the missing values to fill them with categorical values."

Comment (znation) [inline feature-engineering/categorical_imputer.md]: Nevermind, I see below that they are. Resolving.

**Comment by znation** [comment]
Looks good to me. Merge away.

**Pull request merged**